src/LayerPlan.cpp

```diff
@@ -774,6 +774,31 @@ void LayerPlan::writeGCode(GCodeExport& gcode)
                 }
                 gcode.writeTemperatureCommand(prev_extruder, prev_extruder_temp, wait);
             }
+
+            if (!getPrimeTowerIsPlanned())
```

Comment:
But what's happening is that after each extruder change it was moving back to the print and extruding 16mm of filament. Some of that would be the normal post-retract prime but around 10mm was extra. The code I have added makes that big extrude occur before moving back to the print.